### ISSUE TITLE
(Bug) Unable to run Chia using Windows on AMD K10 architecture (AMD Phenom)

### ISSUE THREAD

**Issue by byme1991** [Chia-Network/chia-blockchain#16386]
### What happened?

hello, I would like someone to help me. this is how the chia app works for me on 1.8.2. since I want a stronger version, when I install 2.0 or 2.0.1 I have a problem. the problem is the following. When entering the application, it always says Loading keyring status.... what should I do? thank you

### Version

2.0.0 - 2.0.1

### What platform are you using?

Windows

### What ui mode are you using?

GUI

### Relevant log output

```shell
Loading keyring status
```

**Comment by shaneo257** [comment]
hi, it could mean it's an invalid config.yaml depending on how long it has been spinning. To check you could close Chia, restart your computer, rename the config file, then restart Chia. If it loads up as intended then something is likely wrong with the config after copying over the custom settings

**Comment by ValeryLikavets** [comment]
День добрый. Та же проблема. Система вин 10 и вин 11. Версия 1.8.2 работает. А 2.0.1 никак.

**Comment by newland33** [comment]
same on windows . if i go back on 1.8 it is good.

**Comment by emlowe** [comment]
Putting your logs into DEBUG mode and then providing some log output would be helpful here.
https://docs.chia.net/logging-reference/

Also, trying the CLI instead of the GUI might be helpful , running commands like `chia start wallet`

https://docs.chia.net/check-if-things-are-working

**Comment by newland33** [comment]
> Putting your logs into DEBUG mode and then providing some log output would be helpful here. https://docs.chia.net/logging-reference/
> 
> Also, trying the CLI instead of the GUI might be helpful , running commands like `chia start wallet`
> 
> https://docs.chia.net/check-if-things-are-working

if i do the update , there is no new log entry. if i set to old version log file works fine.

**Comment by wjblanke** [comment]
Can u try rebooting? Also, if logging is not set to DEBUG in config.yaml (it defaults to WARNING) it may take longer for a log file to appear.

**Comment by newland33** [comment]
> Can u try rebooting? Also, if logging is not set to DEBUG in config.yaml (it defaults to WARNING) it may take longer for a log file to appear.

i set debug modus befor update,  3-4 times try do update an read log , but no new log still the old.  today i have set up a new proxmox win11 with tzhe new chia 2.1 .  now it looks good . thx@all

**Comment by ValeryLikavets** [comment]
[1.log](https://github.com/Chia-Network/chia-blockchain/files/13033660/1.log)
[2.log](https://github.com/Chia-Network/chia-blockchain/files/13033661/2.log)

Переустановка на Win11Pro не помогла.
[System_Report.txt](https://github.com/Chia-Network/chia-blockchain/files/13033767/System_Report.txt)

**Comment by wjblanke** [comment]
Valery

2023-10-06T22:43:10.583 wallet chia.rpc.rpc_server        : WARNING  Cannot connect to daemon at ws://localhost:55400
2023-10-06T22:43:10.583 farmer chia.rpc.rpc_server        : WARNING  Cannot connect to daemon at ws://localhost:55400
2023-10-06T22:43:10.587 wallet chia.wallet.wallet_node    : WARNING  Can not connect to keychain at wss://localhost:55400.
2023-10-06T22:43:12.211 full_node chia.rpc.rpc_server     : WARNING  Cannot connect to daemon at ws://localhost:55400

It looks like daemon is not running. Have you tried rebooting? Any other information you can give us about your situation?

**Comment by ValeryLikavets** [comment]
Good afternoon. Whatever I did: rebooted, deleted, installed - nothing helps. I reinstalled different versions of Windows (8,10,11) - only v1,8,2 works on this computer. I also have 7 more computers of different configurations - any version 2.x works on them

**Comment by emlowe** [comment]
This is almost certainly caused by Windows reserving the 55400 port the daemon uses by default.

You can either change the daemon port in config.yaml, or you can modify windows to not reserve this port:
```
dism.exe /Online /Disable-Feature:Microsoft-Hyper-V
reboot
netsh int ipv4 add excludedportrange protocol=tcp startport=55400 numberofports=1
reboot
dism.exe /Online /Disable-Feature:Microsoft-Hyper-V
reboot
```

**Comment by Ilushkin85** [comment]
I have the same problem. On the 1st PC out of 3 does not start. Only 1.8.2 works. Tried all versions of Windows 10.11. - does not start, port 50440- SYN status. What processor do you have installed?

**Comment by Ilushkin85** [comment]
![image](https://github.com/Chia-Network/chia-blockchain/assets/149621069/575f65aa-d0de-4c07-b73a-92cd7d2e79d2)

**Comment by Ilushkin85** [comment]
![image](https://github.com/Chia-Network/chia-blockchain/assets/149621069/04dbe2ad-d121-4aa8-88cb-e65e59a86c64)

**Comment by Ilushkin85** [comment]
Replacing the port with 55301 and so on. it doesn't do anything.

**Comment by Ilushkin85** [comment]
![image](https://github.com/Chia-Network/chia-blockchain/assets/149621069/8d840359-5ad1-4f3c-8932-2befcd5e09c2)

**Comment by ValeryLikavets** [comment]
то есть проблему решить никак?

**Comment by ValeryLikavets** [comment]
HELP! Пулы требуют обновиться до версии 2.X

**Comment by ValeryLikavets** [comment]
![image](https://github.com/Chia-Network/chia-blockchain/assets/58442991/8de3933d-f9cc-4005-a619-0543e0c56ef2)

**Comment by ValeryLikavets** [comment]
неужели все решили проблему? Help!

**Comment by ValeryLikavets** [comment]
dism.exe /Online /Disable-Feature:Microsoft-Hyper-V
reboot
netsh int ipv4 add excludedportrange protocol=tcp startport=55400 numberofports=1
reboot
dism.exe /Online /Disable-Feature:Microsoft-Hyper-V
reboot

It does not help.

**Comment by ValeryLikavets** [comment]
как побороть эту ХРЕНЬ?

**Comment by anonuser66** [comment]
Same thing! Nothing seems to help... PC AMD Phenom X4. Windows 10. Please develpers! Fix this problem ASAP!!!

**Comment by emlowe** [comment]
Everyone with issues in running the GUI - you should try going to the CLI as it makes finding issues easier

`chia run_daemon` - this runs the daemon in the foreground.

then in a second window run

`chia start wallet`

Then please include all logs and console output

**Comment by ValeryLikavets** [comment]
I did everything. But he writes the same things.

[debug.log](https://github.com/Chia-Network/chia-blockchain/files/13368010/debug.log)
[debug__.log](https://github.com/Chia-Network/chia-blockchain/files/13368011/debug__.log)
[debug__1.log](https://github.com/Chia-Network/chia-blockchain/files/13368012/debug__1.log)

**Comment by RutaHunter** [comment]
I hawe same issue, any ideas?

**Comment by RutaHunter** [comment]
Chia can't generate keys.
![Снимок экрана 2023-11-16 103518](https://github.com/Chia-Network/chia-blockchain/assets/66780747/c59769d0-57c0-4f04-abd1-2eb111c13283)

**Comment by anonuser66** [comment]
I will explain in detail the behaviour that I have:
Windows 10, on **any chia 2.x version**.
 I install it normally, when I try to put on the words from my wallet, a little windows message pops up that says: **"start_full_node.exe has stopped working"**. This happens **every time** I install a new 2.x version, the first time I run chia.
Then when I finish entering the words **"Connecting to daemon Attempt XX"** as seen before on this thread.
Of course nothing happens even if I wait an eternity. I close the program and open chia again.
Then the infamous **"Loading keychain status"** appears forever. Doing things to config.yaml file does nothing, changing port does nothing.
Extremily frustrating situation. There is no way I can get it to farm. Please, need avice.
Thanks.

**Comment by leira909089** [comment]
estoy en la misma condicion Loading keyring status AMD Phenon6

**Comment by ValeryLikavets** [comment]
Обидно, что разработчикам пофигу

**Comment by ValeryLikavets** [comment]
Тут есть кто-нибудь?
Is anybody here?
誰かいませんか？
Je, kuna mtu yeyote hapa?
יש פה מישהו?

**Comment by RutaHunter** [comment]
> Тут есть кто-нибудь?
> Is anybody here?

???

**Comment by ValeryLikavets** [comment]
Good afternoon. Just wondering if it's possible to solve this problem. And here there is silence.

**Comment by anonuser66** [comment]
No news or fixes, only a quote:

"Man's nature is not essentially evil. Brute nature has been known to yield to the influence of love. You must never despair of human nature."
-Mahatma Gandhi

**Comment by Ssmith2003** [comment]
> Everyone with issues in running the GUI - you should try going to the CLI as it makes finding issues easier
> 
> `chia run_daemon` - this runs the daemon in the foreground.
> 
> then in a second window run
> 
> `chia start wallet`
> 
> Then please include all logs and console output

Most of use use the GUI because we don't know how to use the CLI. I'm have same problem. Would be more and happy to run in CLI, but can't seem to figure it out. Try paste you command in and does work for me.

**Comment by Ssmith2003** [comment]
2023-12-10T00:22:34.718 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:35.887 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:37.066 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:38.232 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:39.422 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:40.596 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:41.776 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:42.946 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:44.172 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:44.246 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia\rpc\rpc_server.py", line 340, in safe_handle
  File "chia\rpc\rpc_server.py", line 331, in ws_api
  File "chia\rpc\wallet_rpc_api.py", line 531, in get_sync_status
  File "chia\wallet\wallet_node.py", line 161, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2023-12-10T00:22:45.334 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:46.495 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:47.663 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:48.848 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:50.021 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:50.897 wallet wallet_server              : ERROR    Caught exception: RuntimeError: wallet state manager not assigned
Traceback (most recent call last):
  File "chia\util\log_exceptions.py", line 20, in log_exceptions
  File "chia\server\ws_connection.py", line 307, in close
  File "chia\server\server.py", line 555, in connection_closed
  File "chia\wallet\wallet_node.py", line 675, in on_disconnect
  File "chia\wallet\wallet_node.py", line 161, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2023-12-10T00:22:51.168 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:52.345 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:53.580 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:54.317 wallet chia.rpc.rpc_server        : WARNING  Error while handling message: Traceback (most recent call last):
  File "chia\rpc\rpc_server.py", line 340, in safe_handle
  File "chia\rpc\rpc_server.py", line 331, in ws_api
  File "chia\rpc\wallet_rpc_api.py", line 531, in get_sync_status
  File "chia\wallet\wallet_node.py", line 161, in wallet_state_manager
RuntimeError: wallet state manager not assigned

2023-12-10T00:22:54.739 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:55.907 farmer chia.farmer.farmer         : WARNING  No keys exist. Please run 'chia keys generate' or open the UI.
2023-12-10T00:22:56.619 full_node asyncio                 : ERROR    Future exception was never retrieved
future: <Future finished exception=BrokenProcessPool('A process in the process pool was terminated abruptly while the future was running or pending.')>
concurrent.futures.process.BrokenProcessPool: A process in the process pool was terminated abruptly while the future was running or pending.
2023-12-10T00:23:02.998 harvester chia.rpc.rpc_server     : WARNING  Cannot connect to daemon at ws://localhost:55400
2023-12-10T00:23:03.014 wallet chia.wallet.wallet_node    : WARNING  Can not connect to keychain at wss://localhost:55400.
2023-12-10T00:23:03.015 wallet chia.rpc.rpc_server        : WARNING  Cannot connect to daemon at ws://localhost:55400
2023-12-10T00:23:03.030 farmer chia.rpc.rpc_server        : WARNING  Cannot connect to daemon at ws://localhost:55400
2023-12-10T00:23:03.031 farmer chia.farmer.farmer         : WARNING  Can not connect to keychain at wss://localhost:55400.


The port is not port not in  exclusion ranges 
![Screenshot (16)](https://github.com/Chia-Network/chia-blockchain/assets/153409695/c3248d08-961e-48ce-89e2-c81117c1a7e8)
Ever time I change the port in the config file it just lock up loading keyring Status.  
Only way to get it to stop is to delete the config file, but when you load up again just ask for to put you key in again.  
I'm on window 10. it work all good on chia 1.82. didn't become problem to I update to 2.01 and 2.11. I originals installed 2.01 and had problem just went back to 1.82, but know I can't pool on that had to update to new 2.11.  I will be happy to run what you want on CLI, but use the GUI because I only now the very basic.

**Comment by ValeryLikavets** [comment]
Проблему решил - поиенял материнскую плату и процессор на INTEL.

**Comment by Ssmith2003** [comment]
Not sure if translated right, but you update the motherboard and CPU. You talking software or the hardware.

**Comment by RutaHunter** [comment]
> Not sure if translated right, but you update the motherboard and CPU. You talking software or the hardware.
In the chia client version from 2.0.0 there is some incompatibility with some AMD processors. But the developers don’t care, no one is in a hurry to fix it.

**Comment by ValeryLikavets** [comment]
Yes, I changed the hardware platform, and everything worked. And the developers are great ;) don't give a damn about their users.

**Comment by Ssmith2003** [comment]
If that is the answer. I'm likely done with this. I can't go out and get another motherboard.

**Comment by emlowe** [comment]
This issue is because the underlaying BLST library assembler requires SSSE3 (Supplemental SSE3) - this is present in any AMD processor *after* K10, including FX-series and `bobcat` series APUs

It is also in anything from Intel after Core 2 (2006).

**Comment by emlowe** [comment]
Options are being explored as to how to fix it for K10.

**Comment by emlowe** [comment]
We currently believe this is limited to Windows - you might try if WSL to see if it works there

**Comment by ValeryLikavets** [comment]
Well really :). 2 months have passed. At least that's it. I have 4 computers on Phenom x4 - it doesn’t work anywhere. By the way, Windows left the old one without reinstalling. Just a new motherboard and processor.

**Comment by RutaHunter** [comment]
> Options are being explored as to how to fix it for K10.

We hope to resolve the issue as soon as possible. Moreover, everything was fine on older versions of the client.

**Comment by Ssmith2003** [comment]
Have you tried running Linux on a computer with AMD.  If I use Linux will I have to learn all the commands?

**Comment by anonuser66** [comment]
Hello and thanks for all the updates!!
Can anyone confirm that using ubuntu server instead of windows 10 will work? 
I have an AMD Phenom II X4 945

**Comment by emlowe** [comment]
Based on our discussion with the BLST team, we believe Ubuntu will work due to differences in the way the library is compiled. However, we haven't tested it ourselves.
A Ubuntu Live install would be useful here to test

**Comment by wjblanke** [comment]
If you have a K10 CPU and install the new 2.1.2 release, you can copy this version of chia_rs.cp310-win_amd64.pyd over the existing one at c:\users\xxx\AppData\Local\Programs\Chia\resources\app.asar.unpacked\daemon\chia_rs\chia_rs.cp310-win_amd64.pyd. Reboot and restart. This is what I am running on my test machine and it should work fine until we get a fix in for a release (I understand any concerns folks have for running code attached from a github issue)

[chia_rs.cp310-win_amd64.pyd.zip](https://github.com/Chia-Network/chia-blockchain/files/13677877/chia_rs.cp310-win_amd64.pyd.zip)

**Comment by Ssmith2003** [comment]
Thank you very much. I'm back up and running now.

**Comment by wjblanke** [comment]
https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2

**Comment by alexhornet1981** [comment]
En amd phenom II x6 1035 y placa base am3 funciona.

> https://github.com/Chia-Network/chia-blockchain/releases/tag/2.1.4-rc2
> 
> Pruebe esta compilación si tiene problemas

Gracias.

**Comment by Nuke79** [comment]
> Hello and thanks for all the updates!! Can anyone confirm that using ubuntu server instead of windows 10 will work? I have an AMD Phenom II X4 945

Ubuntu server 22.04 + Phenom II 925 - working fine without 2.1.4 fix.

**Comment by alexhornet1981** [comment]
Sí. También funciona.

**Comment by emlowe** [comment]
Fixed in 2.1.4
